File: 0013-roman-to-integer/solution.java
class Solution {
    public int romanToInt(String s) {
        int finalValue = 0;

        Map<Character, Integer> map = new HashMap<>();
        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        for (int i = 0; i < s.length(); i++) {
            char current = s.charAt(i);
            int value = map.get(current);;
            if(i < s.length() - 1) {
                char currentPlusOne = s.charAt(i+1);
                if(
                    (current == 'I' && (currentPlusOne == 'V' || currentPlusOne == 'X')) ||
                    (current == 'X' && (currentPlusOne == 'L' || currentPlusOne == 'C')) ||
                    (current == 'C' && (currentPlusOne == 'D' || currentPlusOne == 'M'))
                ) {
                    value = map.get(currentPlusOne) - map.get(current);
                    i++;
                }
            }
            finalValue += value;
        }

        return finalValue;
    }
}

File: 0013-roman-to-integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        final_value = 0

        map = {
            'I' : 1,
            'V' : 5,
            'X' : 10,
            'L' : 50,
            'C' : 100,
            'D' : 500,
            'M' : 1000
        }

        i = 0

        while (i < len(s)) :
            current = s[i]
            value = map[current]
            if i < (len(s) - 1):
                current_plus_one = s[i+1]
                if (current == 'I' and (current_plus_one == 'V' or current_plus_one == 'X') or \
                    current == 'X' and (current_plus_one == 'L' or current_plus_one == 'C') or \
                    current == 'C' and (current_plus_one == 'D' or current_plus_one == 'M')):
                    value = map[current_plus_one] - map[current]
                    i = i + 1
            final_value = final_value + value
            i = i+1

        return final_value


        
